test/Constraints/patterns.swift

```diff
@@ -314,3 +314,25 @@ switch staticMembers {
 }
 
 _ = 0
+
+// rdar://problem/32241441 - Add fix-it for cases in switch with optional chaining
+
+struct S_32241441 {
+  enum E_32241441 {
+    case foo
+    case bar
+  }
+
+  var type: E_32241441 = E_32241441.foo
+}
+
+func rdar32241441() {
+  let s: S_32241441? = S_32241441()
+
+  switch s?.type {
+  case .foo: // expected-error {{enum case 'foo' not found in type 'S_32241441.E_32241441?'}} {{8-8=.some(}} {{9-9=)}}
```

Comment:
AFAIK, so far, `UnresolvedOriginalExpr` is set only here https://github.com/apple/swift/blob/abd5aa8/lib/Sema/TypeCheckPattern.cpp#L406-L411
It's `UnresolvedMemberExpr`, thus `canAppendPostfixExpression(true) == true`.
I don't think we need to do so for now.